compiler/rustc_lint/src/types.rs

```diff
@@ -1255,23 +1278,45 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
             SpecAbi::Rust | SpecAbi::RustCall | SpecAbi::RustIntrinsic | SpecAbi::PlatformIntrinsic
         )
     }
+
+    /// Find any fn-ptr types with external ABIs in `ty`.
+    ///
+    /// For example, `Option<extern "C" fn()>` returns `extern "C" fn()`
+    fn find_fn_ptr_ty_with_external_abi(&self, ty: Ty<'tcx>) -> Option<Ty<'tcx>> {
```

Comment:
Why does this return a single type? Wouldn't something like `Result<extern "C" fn(), extern "C" fn(FfiUnsafe)>` still cause problems?

Reply:
Fixed!